manifests/init.pp

```diff
@@ -132,6 +137,9 @@
   validate_re($contextpath, ['^$', '^/.*'])
   validate_hash($resources)
   validate_hash($ajp)
+  validate_bool($tomcatNativeSsl)
+  validate_absolute_path($tomcatKeystoreFile)
+  validate_re($tomcatKeystoreType, '^(JKS|JCEKS|PKCS12)$')
```

Comment:
Oh cool :)

Which remindes me, is there a canonical way of validating port numbers?